core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/UniqueIDReadWriteLDAPUserStoreManager.java

```diff
@@ -2185,13 +2186,20 @@ protected void modifyUserInRole(String userNameDN, String groupRDN, int modifyTy
                 logger.debug("User: " + userNameDN + " was successfully " + "modified in LDAP group: " + groupRDN);
             }
         } catch (NamingException e) {
-            String errorMessage =
-                    "Error occurred while modifying user entry: " + userNameDN + " in LDAP role: " + groupRDN;
-            if (log.isDebugEnabled()) {
-                log.debug(errorMessage, e);
+            if (e instanceof NameAlreadyBoundException) {
+                // If concurrent requests bypass the existing user validation check, it can throw an exception
+                // regarding unique key violation. Since the user addition is successful, the exception is ignored.
+                log.warn(String.format("Similar entry found when adding the user to LDAP role: %s. Hence skipping " +
```

Comment:
Shall we have this as a debug log?

Reply:
Check other file as well

Reply:
Fixed with https://github.com/wso2/carbon-kernel/pull/3911/commits/693ca191192f17525d419aa2f30365098de6172f